### DISCUSSION TITLE
Z80 instruction syntax?

### DISCUSSION THREAD

**Discussion by prlaba** [hlorenzi/customasm#102] [Q&A]
I’m building my own “Z80-lite” CPU and am looking for a customizable assembler for the project.  My goal is to come up with an assembler that uses exact Z80 instruction syntax.

customasm looks like a great possibility.  Can anyone familiar with both customasm and the Z80 instruction syntax comment?

One difference I spotted reading through the docs is customasm uses []’s for indirect addressing while the Z80 uses ()’s.  Is it possible to map ()’s into []’s in customasm?

Also, is customasm case insensitive?  For example, if I define a “LD” (load) instruction in custmasm, will it accept “ld” as well?  Or do I need to define both case forms?

Thanks for any insights you might have.

**Comment by hlorenzi** [comment]
You can use either brackets `[]` or parentheses `()` or even mixed `(]` in your mnemonics -- there's no limitation for what they can look like, except for the use of braces `{}`, which are reserved. I think you might have come across the slice syntax `x[7:0]` which is only relevant in expression contexts, e.g. for when you're defining the binary representation for your instructions (not the mnemonics).

All mnemonics are automatically case-insensitive, so there's no need to define them twice. The assembler's built-in directives are also case-insensitive. The only exception are labels, which are case-sensitive, but then of course it's just up to you naming them to your taste.

I think someone else might have worked on a Z80 definition before? I'll let them chime in with any extra information!
